File: lib/rest/yampi/mappers/yampi_installment_mapper.dart
import 'package:sertton/core/checkout/dtos/installment_dto.dart';
import 'package:sertton/rest/types/json.dart';

class YampiInstallmentMapper {
  static InstallmentDto toDto(Json json) {
    return InstallmentDto(
      number: json['installment'] as int,
      value: json['installment_value_formated'] as String,
      totalValue: json['amount_formated'] as String,
      text: json['text'] as String,
      interestFree: (json['tax_value'] as num?) == 0,
    );
  }

  static List<InstallmentDto> toDtoList(Json json) {
    if (json.containsKey('data')) {
      final data = json['data'] as Json;
      if (data.containsKey('installments')) {
        final installments = (data['installments'] as List).cast<Json>();
        return installments.map(toDto).toList();
      }
    }
    return [];
  }
}

File: sertton-react-native-app/src/rest/yampi/mappers/yampi-installment-mapper.ts
import type { Installment } from "@/core/checkout/entities"
import type { YampiInstallment } from "@/rest/yampi/types"

export const YampiInstallmentMapper = () => ({
  toDomain(input: YampiInstallment): Installment {
    return {
      number: input.installment,
      value: input.installment_value_formated,
      totalValue: input.amount_formated,
      text: input.text,
      interestFree: input.tax_value === 0,
    }
  },
})
